threads.go

```diff
@@ -541,11 +551,24 @@ func (t *threads) Subscribe(opts ...options.SubOption) tserv.Subscription {
 
 // putRecord adds an existing record. See PutOption for more.
 func (t *threads) putRecord(ctx context.Context, id thread.ID, lid peer.ID, rec thread.Record) error {
-	knownRecord, err := t.bstore.Has(rec.Cid())
-	if err != nil {
-		return err
+	unknownRecords := []thread.Record{}
+	cid := rec.Cid()
+	for cid.Defined() {
+		exist, err := t.bstore.Has(cid)
+		if err != nil {
+			return err
+		}
+		if exist {
+			break
+		}
+		r, err := t.GetRecord(ctx, id, cid)
+		if err != nil {
+			return err
+		}
+		unknownRecords = append(unknownRecords, r)
+		cid = r.PrevID()
 	}
-	if knownRecord {
```

Comment:
Iterate from received record, while navigating previous ones, up to the point that we recognize one of them (should be the current head).
All the unkonwn records, are stored in a list `unkownRecords`.